src/sources/soundsourceffmpeg.cpp

```diff
@@ -119,98 +228,81 @@ SoundSource::OpenResult SoundSourceFFmpeg::tryOpen(const AudioSourceConfig& /*au
+                << "avformat_find_stream_info() failed and returned"
+                << avformat_find_stream_info_result;
         return OpenResult::FAILED;
     }
 
     //debug only (Enable if needed)
     //av_dump_format(m_pFormatCtx, 0, qBAFilename.constData(), false);
 
-    // Find the first audio stream
-    m_iAudioStream = -1;
-
-    for (unsigned int i = 0; i < m_pFormatCtx->nb_streams; i++)
-        if (m_pFormatCtx->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO) {
-            m_iAudioStream = i;
-            break;
-        }
-
-    if (m_iAudioStream == -1) {
-        qDebug() <<
-                 "SoundSourceFFmpeg::tryOpen: cannot find an audio stream: cannot open"
-                 << localFileName;
-        return OpenResult::FAILED;
-    }
-
-    // Get a pointer to the codec context for the audio stream
-    m_pCodecCtx = m_pFormatCtx->streams[m_iAudioStream]->codec;
-
-    // Find the decoder for the audio stream
-    if (!(m_pCodec = avcodec_find_decoder(m_pCodecCtx->codec_id))) {
-        qDebug() << "SoundSourceFFmpeg::tryOpen: cannot find a decoder for" <<
-                localFileName;
+    // Find and open audio stream for decoding
+    AVStream* pAudioStream = findFirstAudioStream(m_pFormatCtx);
+    if (pAudioStream == nullptr) {
+        qWarning() << "[SoundSourceFFmpeg]"
+                << "No audio stream found";
         return OpenResult::UNSUPPORTED_FORMAT;
     }
-
-    if (avcodec_open2(m_pCodecCtx, m_pCodec, nullptr)<0) {
-        qDebug() << "SoundSourceFFmpeg::tryOpen: cannot open" << localFileName;
-        return OpenResult::FAILED;
+    const OpenResult openAudioStreamResult = openAudioStream(pAudioStream);
+    if (openAudioStreamResult != OpenResult::SUCCEEDED) {
+        return openAudioStreamResult; // early exit on any error
     }
+    m_pAudioStream = pAudioStream;
 
-    m_pResample = std::make_unique<EncoderFfmpegResample>(m_pCodecCtx);
-    m_pResample->openMixxx(m_pCodecCtx->sample_fmt, AV_SAMPLE_FMT_FLT);
-
-    setChannelCount(m_pCodecCtx->channels);
-    setSamplingRate(m_pCodecCtx->sample_rate);
-    setFrameCount((qint64)round((double)((double)m_pFormatCtx->duration *
-                                         (double)m_pCodecCtx->sample_rate) / (double)AV_TIME_BASE));
-
-    qDebug() << "SoundSourceFFmpeg::tryOpen: Sampling rate: " << getSamplingRate() <<
-             ", Channels: " <<
-             getChannelCount() << "\n";
-    if (getChannelCount() > 2) {
-        qDebug() << "ffmpeg: No support for more than 2 channels!";
-        return OpenResult::FAILED;
+    const SINT channelCount = mixxx::getChannelCount(m_pAudioStream);
+    if (channelCount > kMaxChannelCount) {
+        qWarning() << "[SoundSourceFFmpeg]"
+                << "Unsupported number of channels:"
+                << channelCount << ">" << kMaxChannelCount;
+        return OpenResult::UNSUPPORTED_FORMAT;
     }
+    const SINT samplingRate = mixxx::getSamplingRate(m_pAudioStream);
```

Comment:
mixxx::getSamplingRate call looks like you call something that is common for Mixxx, but here it is in the mixxx::(anonymous) namespace. 
Do you have an idea to make it more obvious? e.g introduce a new namespace?